src/Language/PureScript/Scheme/CodeGen/Transpiler.hs

```diff
@@ -216,27 +224,36 @@ caseToScheme values caseAlternatives =
       replaceVariables (exprToScheme result)
                        (variablesToReplace valuesAndBinders)
 
-    -- Each identifier in a VarBinder will be replaced by its corresponding
-    -- value. E.g. `foo m n = m + n' will emit a function whose parameters are
-    -- v and v0, but the `m + n' expression will use `m' and `n' as variable.
-    -- We have to replace each occurrence of `m' with `v' and each occurrence
-    -- of `n' with `v0'.
+
+    variablesToReplace :: [(Expr Ann, Binder a)] -> [(Text, AST)]
     variablesToReplace valuesAndBinders =
-      concatMap go valuesAndBinders
+      concatMap (\(value, binder) -> go (exprToScheme value) binder)
```

Comment:
Why this `exprToScheme`? This probably deserves a comment

Reply:
It is for the same reason we do it at line 171. Should I add a comment nonetheless?

Reply:
Totally, I missed that occurrence too.

Reply:
I tried adding a comment but I couldn't think of anything better than "We are handed with an CoreFn Expr. We have to translate in a Scheme expression." Which is pretty obvious.

What does make it feel weird to you?

Reply:
Right, I guess my confusion is actually more like "why did we not call `exprToScheme` before and now we do"